src/ATen/native/xpu/Loss.cpp

```diff
@@ -73,5 +76,85 @@ Tensor& binary_cross_entropy_backward_out_xpu(
       grad_output, self, target, weight, reduction, grad_input);
 }
 
+static inline at::Tensor apply_loss_reduction(
+    const at::Tensor& unreduced,
+    int64_t reduction) {
+  if (reduction == at::Reduction::Mean) {
+    return unreduced.mean();
+  } else if (reduction == at::Reduction::Sum) {
+    return unreduced.sum();
+  }
+  return unreduced;
+}
+
+Tensor soft_margin_loss_xpu(
```

Comment:
I don't see any reference to `soft_margin_loss_xpu`. Do we still need to register it?